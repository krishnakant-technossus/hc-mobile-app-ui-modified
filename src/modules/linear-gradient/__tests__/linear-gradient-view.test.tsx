// npx jest ./src/modules/linear-gradient/__tests__/linear-gradient-view.test.tsx

import * as React from 'react';
import { render } from '@testing-library/react-native';
import LinearGradientView from '../linear-gradient-view';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';
import Text from '../../text/text';

const gradientColors = [
  {offset: 0, color: '#E4DE74'},
  {offset: 1, color: '#96B660'},
]

describe('LinearGradientView', () => {

  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <LinearGradientView gradientColors={gradientColors}>
            <Text>Test</Text>
          </LinearGradientView>
        </ThemeProvider>,
      );

      const cardItem = wrapper.getByTestId('linear-gradient');

      expect(cardItem).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });
  });

});