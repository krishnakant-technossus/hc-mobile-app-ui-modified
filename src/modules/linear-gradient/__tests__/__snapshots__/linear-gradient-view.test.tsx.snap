// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinearGradientView Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "shadowColor": "#38424D",
        "shadowOffset": Object {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
      },
      undefined,
    ]
  }
  testID="linear-gradient"
>
  <RNSVGSvgView
    bbHeight="100%"
    bbWidth="100%"
    focusable={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <RNSVGGroup>
      <RNSVGDefs>
        <RNSVGLinearGradient
          gradient={
            Array [
              0,
              -1778060,
              1,
              -6900128,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="grad"
          x1="0%"
          x2="0%"
          y1="0%"
          y2="100%"
        />
      </RNSVGDefs>
      <RNSVGRect
        fill={
          Array [
            1,
            "grad",
          ]
        }
        height="100%"
        propList={
          Array [
            "fill",
          ]
        }
        rx={8}
        width="100%"
        x={0}
        y={0}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      Array [
        Object {
          "color": "#1C2530",
          "fontFamily": "System",
          "fontSize": 18,
          "fontStyle": "normal",
          "fontWeight": "400",
          "textAlign": "left",
        },
      ]
    }
    testID="text"
  >
    Test
  </Text>
</View>
`;
